### PR TITLE
Add discussion redirect

### LINKED ISSUE #60: Add `/rfd/0123/discussion` route that redirects to GH PR
It would just be a loader, no page necessary, and do something similar to the RFD view one, except we would not try to fetch the discussion and we would only redirect if the user has the right perms. If the user is not logged in, we should **always** redirect to login (even for public RFDs).

https://github.com/oxidecomputer/rfd-site/blob/c42214b4cda29f84c8d367d0bcfac60295885b55/app/routes/rfd.%24slug.tsx#L82-L121

### PR DESCRIPTION
**Pull request by benjaminleonard**
Fixes #60

### CODE REVIEW

**Comment by benjaminleonard** [comment]
Test link: https://rfd-site-git-rfd-discussion-link-oxidecomputer.vercel.app/rfd/0001/discussion

**Comment by david-crespo** [comment]
I don't understand why it works if I'm not logged in

File: app/routes/rfd.$slug.discussion.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  const num = parseRfdNum(slug)
+  if (!num) throw resp404()
+
+  const user = await isAuthenticated(request)
+
+  const rfd = await fetchRfd(num, user)
+
+  // If someone goes to a private RFD but they're not logged in, they will
+  // want to log in and see it.
+  if (!rfd && !user) throw redirect(`/login?returnTo=/rfd/${num}`)
```
Comment (david-crespo) [inline app/routes/rfd.$slug.discussion.tsx:27]: Oh here's why. You need to change this to an `||`.

**Comment by david-crespo** [comment]
The test failures are because the real data has changed haha. We haven't made a PR here in three months.

**Pull request merged**